tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java

```diff
@@ -0,0 +1,202 @@
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+/**
+ * Integration test for observability of metrics
+ */
+public class MetricsTestCase {
+
+    private ServerInstance serverInstance;
+    private static final String RESOURCE_LOCATION = "src" + File.separator + "test" + File.separator +
+            "resources" + File.separator + "observability" + File.separator + "metrics" + File.separator;
+    private static final String DB_NAME = "TEST_DB";
+    private static Map<String, Pattern> expectedMetrics = new HashMap<>();
+
+    @BeforeTest
+    private void setup() throws Exception {
+        serverInstance = ServerInstance.initBallerinaServer();
+        copyFile(new File(System.getProperty("hsqldb.jar")),
+                new File(serverInstance.getServerHome() + File.separator + "bre" + File.separator + "lib" +
+                        File.separator + "hsqldb.jar"));
+        SQLDBUtils.deleteFiles(new File(SQLDBUtils.DB_DIRECTORY), DB_NAME);
+        SQLDBUtils.initDatabase(SQLDBUtils.DB_DIRECTORY, DB_NAME, "observability/metrics/data.sql");
+        String balFile = new File(RESOURCE_LOCATION + "metrics-test.bal").getAbsolutePath();
+        serverInstance.setArguments(new String[]{balFile, "--observe"});
+        serverInstance.startServer();
+        addMetrics();
+    }
+
+    @Test
+    public void testMetrics() throws Exception {
+        // Test Service
+        Assert.assertEquals(HttpClientRequest.doGet("http://localhost:9090/test").getData(), "Metric Test!");
+        Assert.assertEquals(HttpClientRequest.doGet("http://localhost:9090/product").getData(),
+                "[{\"PRODUCTID\":1,\"PRODUCTNAME\":\"WSO2-IAM\"},{\"PRODUCTID\":3,\"PRODUCTNAME\":\"WSO2-EI\"}]");
+
+        // Send some requests
+        int i = 0;
+        while (i < 5) {
+            HttpClientRequest.doGet("http://localhost:9090/test");
+            HttpClientRequest.doGet("http://localhost:9090/product");
+            i++;
+        }
+
+        // Read the metrics from the prometheus endpoint
+        URL metricsEndPoint = new URL("http://localhost:9797/");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(metricsEndPoint.openConnection()
+                .getInputStream()));
+        reader.lines().filter(s -> !s.startsWith("#")).forEach(line -> {
+            int index = line.lastIndexOf(" ");
+            String key = line.substring(0, index);
+            String value = line.substring(index + 1);
+            Assert.assertTrue(expectedMetrics.get(key).matcher(value).find());
+        });
+
+        reader.close();
+    }
+
+    @AfterClass
+    private void cleanup() throws Exception {
+        serverInstance.stopServer();
+    }
+
+    private static void copyFile(File source, File dest) throws IOException {
+        Files.copy(source.toPath(), dest.toPath(), REPLACE_EXISTING);
+    }
+
+    private static void addMetrics() {
```

Comment:
You can remove the static, and make it as just private method once the expectedMetrics is changed to an instance variable.